xarray/tests/test_backends.py

```diff
@@ -2675,40 +2675,35 @@ def test_write_persistence_modes(self, group) -> None:
             assert_identical(original, actual)
 
     def test_compressor_encoding(self) -> None:
-        original = create_test_data()
         # specify a custom compressor
-
-        if has_zarr_v3 and zarr.config.config["default_zarr_version"] == 3:
-            encoding_key = "codecs"
+        original = create_test_data()
+        if has_zarr_v3 and zarr.config.config["default_zarr_format"] == 3:
+            encoding_key = "compressors"
             # all parameters need to be explicitly specified in order for the comparison to pass below
             encoding = {
+                "serializer": zarr.codecs.BytesCodec(endian="little"),
                 encoding_key: (
-                    zarr.codecs.BytesCodec(endian="little"),
                     zarr.codecs.BloscCodec(
                         cname="zstd",
                         clevel=3,
                         shuffle="shuffle",
                         typesize=8,
                         blocksize=0,
                     ),
-                )
+                ),
             }
         else:
             from numcodecs.blosc import Blosc
 
-            encoding_key = "compressor"
-            encoding = {encoding_key: Blosc(cname="zstd", clevel=3, shuffle=2)}
+            encoding_key = "compressors" if has_zarr_v3 else "compressor"
+            comp = Blosc(cname="zstd", clevel=3, shuffle=2)
+            encoding = {encoding_key: (comp,) if has_zarr_v3 else comp}
```

Comment:
This all feels a bit silly. "compressors" instead of "compressor" and now a tuple instead of a single object.

Reply:
we had to pluralize it to make things work with zarr v3. fwiw I [proposed "compression"](https://github.com/zarr-developers/zarr-python/pull/2463#issuecomment-2558257088) as an arity-ambiguous version, but that wasn't popular enough